src/riak_object.erl

```diff
@@ -505,13 +510,22 @@ merge_acc_to_contents(MergeAcc) ->
     %% `r_content' entries.  by generating their metadata entry and
     %% binary encoding their contents. Bucket Types should ensure this
     %% accumulator only has one entry ever.
-    orddict:fold(fun(_Type, {Meta, CRDT}, {_, Contents}) ->
-                         {riak_kv_crdt:to_mod(CRDT),
-                          [{r_content, riak_kv_crdt:meta(Meta, CRDT),
-                           riak_kv_crdt:to_binary(CRDT)} | Contents]}
-                 end,
-                 {undefined, Keep2},
-                 CRDTs).
+
+    case orddict:size(CRDTs) > 0 of
+        true ->
+            BProps = riak_core_bucket:get_bucket(Bucket),
```

Comment:
This is an issue that jon meredith and I talked about a bunch. There's not good way around it yet, but Riak team is on it... somewhere in the ether.